### PR TITLE
Make link checker ignore links to files added in this PR

### LINKED ISSUE #9628: check-links.js should allow for links that are within the PR
In #9434 a new component is introduced that includes the validator protoascii and the markdown describing how to use it. All component markdown files link to the validator protoascii for reference. However, [check-links.js](https://github.com/ampproject/amphtml/blob/master/build-system/tasks/check-links.js) is showing an error because that link doesn't exist. While technically correct it would be false once the PR is merged.

Potential solutions:
- change check-links to emit a warning instead of an error
- if a deadlink is suspected and it is for a link to `github.com/ampproject/amphtml/blob/master` then check to see if the file exists within the PR

Here's the error
```
[03:46:11] Checking links in extensions/amp-imgur/amp-imgur.md ...
[03:46:11] [✖] https://github.com/ampproject/amphtml/blob/master/extensions/amp-imgur/0.1/validator-amp-imgur.protoascii
[03:46:11] ERROR Possible dead link(s) found in extensions/amp-imgur/amp-imgur.md (please update, or whitelist in build-system/tasks/check-links.js).
[03:46:11] ERROR Possible dead link(s) found. Please update, or whitelist in build-system/tasks/check-links.js. extensions/amp-imgur/amp-imgur.md
```

Note that `extensions/amp-imgur/0.1/validator-amp-imgur.protoascii` is a new file within PR #9434.

### PR DESCRIPTION
**Pull request by rsimha-amp**
Fixes #9628

### CODE REVIEW

File: build-system/tasks/check-links.js [hunk 1]
```diff
@@ -94,6 +99,22 @@ function checkLinks() {
...
+ * Determines if a link points to a file added in the PR.
+ *
+ * @param {string} link Link being tested.
+ * @return {boolean} True if the link points to a file added in the PR.
+ */
+function isLinkToFileAddedInPR(link) {
+  var filesAdded = getStdout(
+      `git diff --name-only --diff-filter=A master...HEAD`).trim().split('\n');
+  var addedInPr = false;
+  filesAdded.forEach(function(file) {
```
Comment (jridgewell) [inline build-system/tasks/check-links.js]: Use a `#some`.

File: build-system/tasks/check-links.js [hunk 2]
```diff
@@ -94,6 +99,22 @@ function checkLinks() {
...
+    if (file.length > 0 && link.includes(file)) addedInPr = true;
```
Comment (jridgewell) [inline build-system/tasks/check-links.js]: What exactly is the test here? What would `link` be?

File: build-system/tasks/check-links.js (hunk 1 above)
Comment (rsimha-amp) [inline build-system/tasks/check-links.js]: Nice idea. Done.

File: build-system/tasks/check-links.js (hunk 2 above)
Comment (rsimha-amp) [inline build-system/tasks/check-links.js]: `link` is any URL found in a `.md` file. For example, `[Foo](https://github.com/ampproject/amphtml/blob/master/contributing/foo.md)`. If `foo.md` were to be added by the same PR, the link checker will flag it as a dead link even though the link will be alive once the PR is merged.

The above function skips links that contain the name of a file being added by the PR.

**Pull request merged**